src/Main.enso

```diff
@@ -1,105 +1,193 @@
+
+## Imports
+# Most Enso files will want to start with importing the standard library.
+# An unqualified import declaration, importing everything from the Base module.
 from Base import all
 
+# Unqualified imports can also be restricted
+# from Base.Network.Http import Header
 
-## the `main` method is the program entry point
-main =
-    # assignments – run the right-hand side and store the result in `operator1`
-    operator1 = 0.up_to 100 . to_vector . map .noise
-    # single-assignment! Can't modify an existing variable
-    # operator1 = 1 – reports a compile error now!
+# Qualified imports bring only the module name into
+# import Base.Network.Http 
+
+## Assignments
+# `=` for assignments, storing the result of the right-hand side under
+# the name on the left.
+x = 1
+y = 10
+result = x + y
+
+# Enso has single assignments: you cannot modify existing variable.
+# x = 2 now will result in a compile error.
+
+## Functions & Methods
+# Declare functions with a name,
+# followed by a space separated list of parameters.
+sum a b = a + b
+result2 = sum x y
+
+# When declared in the top-level, functions are methods of the current module.
+# Methods can be called with `.`, and `here` refers to the current module.
+# `sum` is a method of the current module, so can be called on `here`.
+result3 = here.sum 1 2
+# Methods can also be declared on a type, see Type Definition below.
 
-    # Method calls – `.` calls method `sort` of `operator1`
-    # We can pass additional arguments
-    operator1.sort Ascending
+# Functions can be declared with default values for parameters.
+product x=1 y=2 = x * y
 
-    # Operators have different precedence depending on spacing
+# They can then be called with default arguments by name
+result3 = product x=20 y=100 # equivalent to product 20 100
+result4 = product x=20 # equivalent to product 20 2
+result5 = product 20 # equivalent to product 20 2
+result6 = product 20 ... # returns a function expecting a parameter for y
+
+# The entry point for an Enso program is a method `main` in `Main.enso`.
+main =
+    # Methods can be called with `.`.
+    # Methods can be passed additional arguments.
+    # `up_to` creates a `Range` from 1 up to 10.
+    range = 1.up_to 10
+
+    # Operators, including `.`, have different precedence depending on spacing
     # No-spaces binds stronger than a spaced version
     # foo bar . baz ==> (foo bar).baz
     # foo bar.baz   ==> foo (bar.baz)
-    operator1 = 0.up_to 100 . to_vector . map .noise
+    sorted_list = range . to_vector . sort Ascending
+    # equivalent to:
+    # sorted_list = (range.to_vector).sort Ascending
 
-    # Operator sections!
+    # Operator sections:
     # (+ 5) => a function that takes x and returns x+5
-    # (5 +) => ...
+    # (5 +) => as above.
     # [1, 2, 3] . map (+ 5)
     # [1, 2, 3] . map +5
     # .noise => a function that takes x and return x.noise
-    operator1.sort
+    sorted_list.sort  # a function expecting a comparator to sort on.
```

Comment:
This doesn't really make sense – `sort` has a bunch of defaulted argument, so this just sorts the sorted list again.